src/atdgen_codec_runtime.ml

```diff
@@ -173,7 +173,35 @@ struct
     else
       Some (decode json)
 
-  let fieldOptional s f = optional (field s f)
+  (* Define our own version of field which returns None if key is not found *)
+  let custom_field key decode json =
```

Comment:
```suggestion
  let fieldOptional key decode json =
```

and custom_optional is not needed then

Reply:
Changed in https://github.com/ahrefs/bs-atdgen-codec-runtime/pull/11/commits/0549888c3e2635b4876a9c4a5239ae0670552712